docs/csi-snapshot-import.md

```diff
@@ -0,0 +1,79 @@
+# Import snapshot into Kubernetes
+
+## Use Cases
+- Reinstall Kubernetes or migrate snapshot from one Kubernetes cluster to another
+- Disaster recovery from backup snapshot
+- Containerize legacy application
+
+## Import Guidance
+
+#### Prerequisition
+>Check [examples/snapshotimport/import-pvc.yaml](./examples/snapshotimport/import-pvc.yaml), [examples/snapshotimport/import-vol.yaml](./examples/snapshotimport/import-vol.yaml), and [examples/snapshotimport/mount-pvc-to-pod.yaml](./examples/snapshotimport/mount-pvc-to-pod.yaml)
+
+Importing a snapshot requires its source volume object already exists in Kubernetes. That means the volume object should either be created inside Kubernetes, or have already been imported to Kubernetes.
+For importing the source volume:
+
+1. Follow [csi-volume-import.md](./csi-volume-import.md) to create the PVC object, and
+2. Make sure you have mounted the PVC to a pod. This is important because only after that, the PVC object is available to snapshotter.
+
+#### Create Snapshot Objects
+>Check [examples/snapshotimport/backend-content.yaml](./examples/snapshotimport/backend-content.yaml) and [examples/snapshotimport/backend-snapshot.yaml](./examples/snapshotimport/backend-snapshot.yaml)
+
+When you have the source volume ready as prescribed above, follow the steps below to create `VolumeSnapshotContent` and `VolumeSnapshot` objects manually:
+
+1. Create and deploy a snapshot content object with `snapshotHandle` configured to be the **full** name of the snapshot in backend, and `volumeSnapshotRef` to be the volume snapshot at your choice.
+
+   Example file `backendcontent.yaml`:
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshotContent
+metadata:
+  name: backend-content
+spec:
+  deletionPolicy: Delete
+  driver: pure-csi
+  source:
+    # TODO: change this to your backend snapshot name.
+    # Make sure to put the full name here. 
+    snapshotHandle: test-backend-vol.test-backend-snapshot
+  volumeSnapshotRef:
+    # TODO: Put your own snapshot object name here.
+    # Make sure it's identical with the name in "backendsnapshot.yaml"
+    name: backend-snapshot
+    namespace: default
+```
+  Note: the `snapshotHandle` is the full name of the backend snapshot. It can be:
+   * With no namespace: test-backend-vol.test-backend-snapshot
```

Comment:
Change `namespace` for something else, eg `cluster` here and the two bullet points below